dtreeviz/shadow.py

```diff
@@ -230,17 +231,23 @@ def get_node_type(_tree_model):
         return node_type
 
     @staticmethod
-    def get_leaf_sample_counts(_tree_model):
+    def get_leaf_sample_counts(_tree_model, min_samples=0, max_samples=maxsize):
```

Comment:
thanks @parrt for the review. I will try to solve it tomorrow, or in the beginning of the next week. I'm out of laptop in this weekend :)

Reply:
hi @parrt. I changed a little the code... now maxsize is calculated from max of node samples and the return value is a tuple of two numpy array.

personally, I would let the filter logic into get_leaf_sample_counts function, because I will like the viz_leaf_samples function to have a single scope (that of making a plot based on some input). This will help in the future whit unit testing, etc.

Reply:
I also wrote a filtering logic based only on numpy, but it doesn't seem very readable.

```@staticmethod
     def get_leaf_sample_counts(_tree_model, min_samples=0, max_samples=None):
         node_type = ShadowDecTree.get_node_type(_tree_model)
         n_node_samples = _tree_model.tree_.n_node_samples
    
         leaf_index = np.where(node_type)[0]
         leaf_node_samples = n_node_samples[leaf_index]
         max_samples = max_samples if max_samples else leaf_node_samples.max()
         leaf_index = leaf_index[(leaf_node_samples >= min_samples) & (leaf_node_samples <= max_samples)]
         leaf_node_samples = leaf_node_samples[(leaf_node_samples >= min_samples) & (leaf_node_samples <= max_samples)]
         return leaf_index, leaf_node_samples
```

Reply:
i suppose that if it's too big to do with plain python, it's too big to display ;)